test/manipulation.test.js

```diff
@@ -889,6 +897,51 @@ describe('manipulation', function() {
     });
   });
 
+  describe('updateOrCreate when forceId is true', function() {
+    var Post;
+    before(function definePostModel(done) {
+      var ds = getSchema();
+      Post = ds.define('Post', {
+        title: {type: String, length: 255},
+        content: {type: String},
+      }, {forceId: true});
+      ds.automigrate('Post', done);
+    });
+
+    it('fails when id does not exist in db', function(done) {
+      var post = {id: 123, title: 'a', content: 'AAA'};
+      Post.updateOrCreate(post, function(err) {
+        err.statusCode.should.equal(404);
+        done();
+      });
+    });
+
+    it('works on create if the request does not include an id', function(done) {
+      var post = {title: 'a', content: 'AAA'};
+      Post.updateOrCreate(post, function(err, p) {
+        if (err) return done(err);
+        p.title.should.equal(post.title);
+        p.content.should.equal(post.content);
+        done();
+      });
+    });
+
+    it('works on update if the request includes an existing id in db', function(done) {
+      Post.create({title: 'a', content: 'AAA'},
+      function(err, post) {
```

Comment:
Can you move this to the previous line?

```js
Post.create({title: 'a', content: 'AAA'}, function(err, post) {
```

Also since this is on `master` which supports only Node.js 4 and newer, it's usually better to use an arrow function (same applies to the other newly added tests):

```js
Post.create({title: 'a', content: 'AAA'}, (err, post) => {
```

FWIW, I would personally use Promises instead, but I understand it may be better to keep consistency with the existing callback-based coding style in this file.